tableprinter/tableprinter.go

```diff
@@ -0,0 +1,86 @@
+package tableprinter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/fluidkeys/fluidkeys/colour"
+)
+
+const gutterWidth = 2
+
+type Row struct {
+	columns []string
+}
+
+type TablePrinter struct {
+	headers Row
+	rows    []Row
+}
+
+func (tp *TablePrinter) PushRow(columnCells Row) {
+	tp.rows = append(tp.rows, columnCells)
+}
+
+func (tp *TablePrinter) PushHorizontalUnderlines() {
```

Comment:
I'm not sure why this is the public interface, surely it's the `TablePrinter`'s job to determine when to push horizontal underlines, depending on the rows I've pushed?